torch/csrc/autograd/python_ir.cpp

```diff
@@ -0,0 +1,126 @@
+#include "torch/csrc/autograd/python_ir.h"
+#include "torch/csrc/utils/python_strings.h"
+
+using namespace torch::autograd;
+
+PyObject* THPNodeClass = nullptr;
+
+PyObject* THPNode_Wrap(const std::shared_ptr<Node>& node)
+{
+  if (!node) {
+    Py_RETURN_NONE;
+  } else {
+    auto type = (PyTypeObject*) THPNodeClass;
+    THPNode* obj = (THPNode*)type->tp_alloc(type, 0);
+    if (obj) {
+      obj->cdata = node;
+    }
+    return (PyObject*) obj;
+  }
+}
+
+static int THPNode_traverse(THPNode *self, visitproc visit, void *arg)
+{
+  if (self->cdata) {
+    if (auto fn = dynamic_cast<PyNode*>(self->cdata.get())) {
+      Py_VISIT(fn->pyobj);
+    }
+  }
+  return 0;
+}
+
+static int THPNode_clear(THPNode *self)
+{
+  if (self->cdata) {
+    if (auto fn = dynamic_cast<PyNode*>(self->cdata.get())) {
+      fn->pyobj = nullptr;
+    }
+  }
+  self->cdata.reset();
+  return 0;
+}
+
+static void THPNode_dealloc(THPNode* self)
+{
+  PyObject_GC_UnTrack(self);
+  THPNode_clear(self);
```

Comment:
@apaszke So, It looks like I cargo-culted this code from THPVariable_dealloc. But I guess indeed we should not have a clear here; the clear is necessary for THPVariable because there is only at most one Python object per Variable, but that invariant doesn't hold here.